src/components/v2/devtronStackManager/DevtronStackManager.type.ts

```diff
@@ -0,0 +1,180 @@
+import { ResponseType } from '../../../services/service.types'
+import { ActionResponse } from '../../external-apps/ExternalAppService'
+
+export enum ModuleStatus {
+    HEALTHY = 'healthy',
+    INSTALLED = 'installed',
+    INSTALLING = 'installing',
+    INSTALL_FAILED = 'installFailed',
+    NONE = 'none',
+    NOT_INSTALLED = 'notInstalled',
+    TIMEOUT = 'timeout',
+    UNKNOWN = 'unknown',
+    UPGRADING = 'upgrading',
+    UPGRADE_FAILED = 'upgradeFailed',
+}
+
+export enum ModuleActions {
+    INSTALL = 'install',
+    UPGRADE = 'upgrade',
+}
+
+export interface StackDetailsType {
+    isLoading: boolean
+    discoverModulesList: ModuleDetails[]
+    installedModulesList: ModuleDetails[]
+    releaseNotes: ReleaseNotes[]
+    logPodName: string
+    errorStatusCode: number
+}
+
+export interface StackManagerNavItemType {
+    installedModulesCount: number
+    installationStatus: ModuleStatus
+    currentVersion: string
+    newVersion: string
+}
+
+export interface StackManagerNavLinkType {
+    name: string
+    href: string
+    icon: React.FunctionComponent<React.SVGProps<SVGSVGElement>>
+    className: string
+}
+
+export interface StackManagerPageHeaderType {
+    detailsMode: string
+    selectedModule: ModuleDetails
+    handleBreadcrumbClick: () => void
+}
+
+export interface ModuleDetails {
+    id: string
+    name: string
+    info: string
+    icon: string
+    installationStatus: ModuleStatus
+    baseMinVersionSupported?: string
+}
+
+export interface ModuleListingViewType {
+    modulesList: ModuleDetails[]
+    currentVersion: string
+    isDiscoverModulesView?: boolean
+    handleModuleCardClick: (moduleDetails: ModuleDetails, fromDiscoverModules: boolean) => void
+    history: any
+}
+
+export interface ModuleDetailsViewType {
+    moduleDetails: ModuleDetails
+    setDetailsMode: React.Dispatch<React.SetStateAction<string>>
+    setShowManagedByDialog: React.Dispatch<React.SetStateAction<boolean>>
+    serverInfo: ServerInfo
+    upgradeVersion: string
+    logPodName?: string
+    fromDiscoverModules?: boolean
+    history: any
+    location: any
+}
+
+export interface ModuleInstallationStatusType {
+    installationStatus: ModuleStatus
+    appName?: string
+    logPodName?: string
+    isUpgradeView?: boolean
+    upgradeVersion?: string
+    latestVersionAvailable: boolean
+}
+
+export interface InstallationWrapperType {
+    moduleName?: string
+    installationStatus: ModuleStatus
+    logPodName?: string
+    serverInfo: ServerInfo
+    upgradeVersion: string
+    isUpgradeView?: boolean
+    setShowManagedByDialog: React.Dispatch<React.SetStateAction<boolean>>
+    history: any
```

Comment:
Can we use the type of history and location

Reply:
Sure, will try.

Reply:
Done